### ISSUE TITLE
Update Dataset concepts with warning about production use

### ISSUE THREAD

**Issue by Vyrastas** [openthread/ot-docs#158]
There was info added to the DATASET readme in the main repo https://github.com/openthread/openthread/pull/10026/files#diff-38fb58c06769c0283e1dd05c06244d63d29561169ca8a2b56166130255467a04

Specifically about restrictions for dataset CLI commands in production.  We should add a similar warning to our Dataset concepts guide: https://github.com/openthread/ot-docs/blob/main/site/en/reference/cli/concepts/dataset.md

**Comment by EskoDijk** [comment]
Yes! Background: dataset CLI commands are designed to be powerful, and for use in testing (including testing fault situations that would otherwise rarely occur). They allow breaking a Thread Network's active or pending datasets, leading to out-of-spec behavior of nodes.

At the same time, vendors may typically include CLI commands in their Thread product as an API between application and stack. So this allows careless applications to accidentally break a user's Thread network.
